imap_processing/lo/l0/data_classes/science_direct_events.py

```diff
@@ -0,0 +1,197 @@
+
+    Bit Shifting:
+    TOF0, TOF1, TOF2, TOF3, and CKSM all must be shifted by one bit to the
+    left. All other fields do not need to be bit shifted.
+
+    The raw values are computed for L1A and will be converted to
+    engineering units in L1B.
+
+    Attributes
+    ----------
+    SHCOARSE : int
+        Spacecraft time.
+    COUNT: int
+        Number of direct events.
+    DATA: str
+        Compressed TOF Direct Event time tagged data.
+    TIME: numpy.ndarray
+        time tag for the direct event
+    ENERGY: numpy.ndarray
+        energy of the direct event ENA.
+    MODE: numpy.ndarray
+        Indication of how the data is packed.
+    TOF0: numpy.ndarray
+        Time of Flight 0 value for direct event.
+    TOF1: numpy.ndarray
+        Time of Flight 1 value for direct event.
+    TOF2: numpy.ndarray
+        Time of Flight 2 value for direct event.
+    TOF3: numpy.ndarray
+        Time of Flight 3 value for direct event.
+    CKSM: numpy.ndarray
+        This is checksum defined relative to the TOFs
+        condition for golden triples. If golden triples are below
+        a certain threshold in checksum it's considered golden, otherwise,
+        it's considered a silver triple. This is important for the compression
+        for golden triples because it's used to recover TOF1 because
+        compression scheme to save space on golden triples doesn't send
+        down TOF1 so it's recovered on the ground using the checksum
+    POS: numpy.ndarray
+        Stop position for the direct event. There are 4 quadrants
+        on the at the stop position.
+
+    Methods
+    -------
+    __init__(packet, software_vesion, packet_file_name):
+        Uses the CCSDS packet, version of the software, and
+        the name of the packet file to parse and store information about
+        the Direct Event packet data.
+    """
+
+    SHCOARSE: int
+    COUNT: int
+    DATA: str
+    TIME: np.ndarray
+    ENERGY: np.ndarray
+    MODE: np.ndarray
+    TOF0: np.ndarray
+    TOF1: np.ndarray
+    TOF2: np.ndarray
+    TOF3: np.ndarray
+    CKSM: np.ndarray
+    POS: np.ndarray
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        """Initialize Science Direct Events Data class.
+
+        Parameters
+        ----------
+        packet: dict
+            single packet from space_packet_parser.
+        software_version: str
+            current version of IMAP-Lo processing.
+        packet_file_name: str
+            name of the CCSDS file where the packet
+            originated.
+        """
+        super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
+        self.set_attributes(packet)
+        # TOF values are not transmitted for certain
+        # cases, so these can be initialized to the
+        # CDF fill val and stored with this value for
+        # those cases.
+        self.TIME = np.ones(self.COUNT) * GlobalConstants.DOUBLE_FILLVAL
```

Comment:
Do these need to be a specific datatype like "uint8" rather than "float"?

As an aside and ignored for this PR: Another thought I had after looking at some of this repetition is if we should think about Structured/named Arrays instead of dataclasses?
https://numpy.org/doc/stable/user/basics.rec.html#
So then everything would be in a single numpy array, but lookup via the name. `self.data["energy"]` for example.